### ISSUE TITLE
Accessibility guidance

### ISSUE THREAD

**Issue by sarawilcox** [nhsuk/nhsuk-service-manual-community-backlog#449]

**Comment by sarawilcox** [comment]
Some info from other organisations:

- US piece:[ WHY ACCESSIBILITY WIDGETS MAY NOT HELP PREVENT ADA  LAWSUITS](https://www.samuelandstein.com/why-accessibility-widgets-may-not-help-prevent-ada-lawsuits/)

**Comment by sarawilcox** [comment]
Example:[ Parliamentary and Health Service Ombudsman](https://www.ombudsman.org.uk/)

![Screenshot 2022-10-14 at 16 32 08](https://user-images.githubusercontent.com/45792959/195894161-0b34b093-e464-42b7-8b8c-ad0ab03ba047.png)

**Comment by sarawilcox** [comment]
See also the Overlay factsheet re accessibility overlays, with signatures from many accessibility experts: https://overlayfactsheet.com/

> "No overlay product on the market can cause a website to become fully compliant with any existing accessibility standard and therefore cannot eliminate legal risk."

**Comment by sarawilcox** [comment]
We're seeing examples of organisations applying accessibility widgets or overlays, thinking that they resolve accessibility issues but they can cause more issues than they solve. We're keen to source more evidence around this.

**Comment by sarawilcox** [comment]
Multiple overlays

![Screenshot 2022-11-10 at 15 49 18](https://user-images.githubusercontent.com/45792959/201141564-7d94d03d-8a78-4af6-a97f-6f689aefe57a.png)

**Comment by sarawilcox** [comment]
Articles: 
- Spindogs: [A Guide to Accessibility Overlays & Digital Accessibility](https://www.spindogs.co.uk/blog/a-guide-to-accessibility-overlays-digital-accessibility/)
- Timothy Springer (on LinkedIn): [Is Deploying an Accessibility Overlay or Widget Better Than Doing Nothing?](https://www.linkedin.com/pulse/deploying-accessibility-overlay-widget-better-than-doing-springer/)

**Comment by albfreeman** [comment]
I've tested the [Mapesbury Medical Group](https://mapesburymedicalgroup.co.uk/) using Dragon dictation software to navigate the website using voice. The overlays and accessibility toolbar caused me some problems. 

When I want to activate a link or a button on a webpage, I say "click" followed by the visible name of the link or button. If the name of a link or button isn't obvious to me (this sometimes happens if the button is an unlabelled image or graphic) I say "click link" or "click button". Dragon then presents me with numbered options overlayed next to every link or button on the page. I then just say "choose X", where X is the number of the link or button I want to activate. The whole process takes just a couple of seconds. It's no slower than using a mouse.

However, the Mapesbury website, with its multiple overlays and its accessibility widget, doesn't work well when controlled by voice. 

I first need to dismiss the coronavirus overlay. The first thing I do when I see something like this is say "click close". If the close icon in the corner of the overlay has been coded well, this command should have the effect of clicking the close icon and dismissing the overlay. This doesn't work on this website though. I then tried my next option and said "click link" or "click button". But I wasn't presented with any numbered options at all, never mind the one I really want (the cross to dismiss the overlay). In fact, if I say "click link" the whole page scrolls down a bit (this is odd. I've not seen this happen on other websites). Finally, I tried saying "press escape". This is the same as pressing the Escape key on a keyboard and this can sometimes close an overlayed element that has the focus. This did dismiss the coronavirus overlay.

Next I wanted to dismiss the "contact your GP practice online" overlay, and this was challenging in different ways. Again, "click close" didn't do anything. If I said "click button", this actually activated the "ReachDeck" accessibility toolbar at top the page (again this is strange and I've not seen this behaviour elsewhere). I tried saying "click link", but this just showed me the clickable links on the page behind the overlay. But the numbers were not in the correct order (they are usually in a logical sequence) and links immediately behind the overlay were not available. Whichever links were numbers 5 and 9 to 14 weren't visible at all, presumably because of the overlay. This is shown in the attached screenshot where the available numbered links are 1...3...2...4...6...7...8...15...16...etc.

![Mapesbury MG Dragon click button options](https://user-images.githubusercontent.com/113704643/201681127-febb83ec-9e91-430d-b72e-8d5ab3cfc7ba.png)

I also tried selecting some of the options on the overlay using Dragon, rather than just trying to dismiss it. But the links and buttons on the overlay don't work properly either. So, not only is there no way for somebody using Dragon to dismiss the contact overlay, it's also not possible to use it if I wanted to.

A last resort when using Dragon to click a link is what it calls its "MouseGrid". If I say "mouse grid", the MouseGrid displays a three-by-three grid over the screen with the sections numbered one to nine. Choosing a number, puts another smaller grid in that part of the screen. I keep saying numbers as the grid gets progressively smaller, focused where I want the cursor to be, and the mouse cursor eventually ends up where I want to click. I had to do this to dismiss the GP contact overlay, saying "mouse grid… 1… 3… 4… 8… Click" to eventually dismiss the contact overlay.

So, as a user of Dragon, it took me several minutes to get to the website behind the overlays, at which point I was then able to more easily select links in the main site content.

As regards the accessibility widget itself, this isn't useful to me as somebody with an accessibility issue. It's impossible for something like that to cater for every accessibility need. It could be useful for people who have problems viewing and reading the content, but for somebody like me who needs to be able to control a page using voice, that sort of tool doesn't help. And as I explained above, it actually caused the problem when I unintentionally activated it while trying to dismiss and overlay. 

I've also heard mixed things about the benefits of on-site accessibility widgets like that. For example, people who need to be able to "hear" the contents of websites will often already have their own solution, such as VoiceOver or JAWS. Whatever solution an individual person uses, they will use it on every website they visit, rather than only being able to use websites that have custom widgets. I think health providers should prioritise having well-structured, well-designed and well-written websites that help all users before they add accessibility widgets that only help some people.

**Comment by sarawilcox** [comment]
### Accessibility widgets and overlays 

We are proposing to put some guidance to senior accessibility leads as follows.

#### Do not use accessibility overlays or accessibility widgets 

These can create problems for people when they get tapped accidentally or when they cover up information. People find them confusing and unhelpful. 

People with access needs often use their preferred assistive technologies across all websites. Widgets only meet some needs and only deal with one part of a user journey. They cannot hand over from one system or site to another. What helps people with access needs more are well-designed, well-structured websites that are:
- built from the bottom up with accessibility-tested code and designs, like the NHS frontend and design system
- tested with users with access needs

Accessibility widgets cannot make a website fully compliant with accessibility standards or protect you from legal action.

**Comment by sarawilcox** [comment]
We're also looking at writing some guidance on banners and banner ads. This includes:
- overlay banners that sit on top of websites and get in the way of user journeys 
- banner images or cards for campaigns, which often include images with text in them and don't include alt text - the messaging would be better integrated into content at the most relevant part of a user journey

The guidance will probably need to distinguish urgent/necessary banners (like the [emergency alert](https://github.com/nhsuk/nhsuk-service-manual-community-backlog/issues/149)) and unnecessary clutter.

**Comment by sarawilcox** [comment]
[Why accessibility overlays do not improve site accessibility](https://business.scope.org.uk/article/why-accessibility-overlays-and-widgets-do-not-improve-your-website-accessibility/)

**Comment by sarawilcox** [comment]
We've run the widgets guidance by internal senior accessibility leads and some accessibility experts. We're now iterating the draft guidance for the service manual.

**Comment by anupshah** [comment]
Hope you don't mind me weighing in on this also - I am very supportive of your findings here. I work for a PEP and we've seen tenders requesting accessibility widgets/toolbars/overlays (i.e. a solution rather than stating the problem) and have been pointed to some suppliers who have been using them on some NHS web sites.

But years of personal experience, research and conversations with experts on this I am in agreement with you, and have also tried to push back for the reasons you have stated.

Many years ago, there was a great, researched article on this which I sadly cannot find currently, but while trying to find it, came across these which are useful, and some are from reasonably popular resources:

- https://blog.usablenet.com/inclusive-accessibility-crushes-widgets-and-overlays
- https://www.accessibility.works/blog/avoid-accessibility-overlay-tools-toolbar-plugins/
- https://business.scope.org.uk/article/why-accessibility-overlays-and-widgets-do-not-improve-your-website-accessibility
- https://www.levelaccess.com/blog/the-many-pitfalls-of-accessibility-overlays

Anecdotally as well, looking at some recently I found that they were not always fully accessible themselves (e.g. often icons only without text, and we know icons only usually does not test very well).

Others had an impressive array of features (so can see why tenders may be wowed by them). However, that left me thinking that these are features you get at the operating system/device or browser level, or you have specialised software if you need such capabilities, which creates consistency for your entire online experience not just one web site at a time where you have to learn different methods.
